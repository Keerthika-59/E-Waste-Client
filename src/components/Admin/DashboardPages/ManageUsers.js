import React, { useState, useEffect } from 'react'
import APIHelper from '../../API/apihelper'
import { Card,Table,Button } from 'react-bootstrap'

const ManageUsers = () => {

    const [users, setUsers] = useState([{
        name: "",
        phoneNumber: "",
        email: "",
        gender: "",
        password: "",
        city: "",
        address: ""
    }]);

    useEffect(() => {
        fetch(`${APIHelper.API_URL}`).then(res => {
            if (res.ok) {
                return res.json()
            }
        }).then(jsonRes => setUsers(jsonRes))
    },[])

    function removeUser(email){
        alert(`removed ${email}`)
    }
    // console.log(`users ${users}`)
    return (
        <>
         <Table striped bordered hover>
                <thead>
                    <tr>
                        <th>#</th>
                        <th>Name</th>
                        <th>Phone Number</th>
                        <th>E-mail</th>
                        <th>City</th>
                        <th>Address</th>
                        <th>Action</th>
                    </tr>
                </thead>
                <tbody>
                {users.map(user =>
                    <tr>
                        <td>{user.id}</td>
                        <td>{user.name}</td>
                        <td>{user.phoneNumber}</td>
                        <td>{user.email}</td>
                        <td>{user.city}</td>
                        <td>{user.address}</td>
                        {/* <td><Button style={{backgroundColor:"red",margin:"10px"}}>DELETE</Button></td>     */}
                        {/* {this.removeToCollection(key, e)} */}
                        <td> <Button variant="danger"  onClick={() => {if(window.confirm('Do you want to delete the user?')){removeUser(user.email)};}}>DELETE</Button></td>
                        
                    </tr>)}
                   
                </tbody>
            </Table>
        </>
    )
}


export default ManageUsers
