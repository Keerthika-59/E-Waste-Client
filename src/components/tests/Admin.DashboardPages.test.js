import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route,} from 'react-router-dom'
import ManageMessagaes from '../Admin/DashboardPages/ManageMessages'
import ManageUsers from '../Admin/DashboardPages/ManageUsers';
import PaginationPage from '../Admin/DashboardPages/PaginationPage';

test('renders ManageMessagaes', () => {
    const div=document.createElement("div");
    <Route>
        ReactDom.render(<ManageMessagaes/>,div);
    </Route>
    ReactDOM.unmountComponentAtNode(div);
   });
   test('should test ManageUsers component', () => {
    const wrapper = shallow(<ManageUsers />);
    expect(wrapper).toMatchSnapshot();
   });
   test('should test PaginationPage component', () => {
    const wrapper = shallow(<PaginationPage />);
    expect(wrapper).toMatchSnapshot();
   });