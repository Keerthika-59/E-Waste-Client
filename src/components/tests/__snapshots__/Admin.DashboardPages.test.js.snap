// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test ManageUsers component 1`] = `
<ForwardRef
  bordered={true}
  hover={true}
  responsive={true}
  striped={true}
>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th>
        Name
      </th>
      <th>
        Phone Number
      </th>
      <th>
        E-mail
      </th>
      <th>
        City
      </th>
      <th>
        Address
      </th>
      <th>
        Action
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td />
      <td />
      <td />
      <td />
      <td />
      <td />
      <td>
         
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          variant="warning"
        >
          DELETE
        </Button>
      </td>
    </tr>
  </tbody>
</ForwardRef>
`;

exports[`should test PaginationPage component 1`] = `
<Pagination
  aria-label="Page navigation example"
  className="paginationcss"
  listTag="ul"
  tag="nav"
/>
`;
